client/app/(auth)/signUp.tsx

```
@@ -0,0 +1,282 @@
+import {
+  getAuth,
+  createUserWithEmailAndPassword,
+  sendEmailVerification,
+} from "@react-native-firebase/auth";
+import { router } from "expo-router";
+import { Eye, EyeSlash, WarningCircle, X } from "phosphor-react-native";
+import { useState } from "react";
+import { Controller, useForm } from "react-hook-form";
+import {
+  View,
+  Text,
+  KeyboardAvoidingView,
+  Keyboard,
+  TouchableWithoutFeedback,
+  TextInput,
+  Pressable,
+} from "react-native";
+import * as yup from "yup";
+import GoogleLogo from "@/assets/icons/google.svg";
+import Spinner from "@/components/Spinner";
+import { useAppDispatch } from "@/state/hooks";
+import { setUser } from "@/state/slices/userSlice";
+import { LoginProvider } from "@/types/providers";
+import { signInWithGoogle } from "@/utils/socialAuth";
+
+const schema = yup.object({
+  email: yup
+    .string()
+    .required("Email is required")
+    .email("Enter a valid email address"),
+  password: yup
+    .string()
+    .required("Password is required")
+    .min(8, "Password must be at least 8 characters"),
+});
+
+type SignUpFormData = {
+  email: string;
+  password: string;
+};
+
+const SignUp = () => {
+  const [hidePassword, setHidePassword] = useState(true);
+  const [errorMessage, setErrorMessage] = useState("");
+  const [loading, setLoading] = useState(false);
+  const dispatch = useAppDispatch();
+  const auth = getAuth();
+
+  const {
+    control,
+    handleSubmit,
+    formState: { errors, isValid },
+    setError,
+  } = useForm<SignUpFormData>({
+    resolver: yupResolver(schema),
+    defaultValues: {
+      email: "",
+      password: "",
+    },
+    mode: "onSubmit",
+  });
+
+  const signUpWithProvider = async (provider: LoginProvider) => {
+    try {
+      if (provider === LoginProvider.Google) {
+        const { user } = await signInWithGoogle();
+        dispatch(
+          setUser({
+            userId: user.uid,
+            email: user.email ?? undefined,
```

Is this ever supposed to be undefined? If it's possible, should we throw an error or at least place this in an if check similar to what we do for signing up with username and password on line 92? Actually though, we also do this null coalesce on line 104, so we should also follow whatever we do here there

In my opinion if for some reason we don't get an email back from the server it should be an error, that doesn't make sense

---

Yeah it shouldn't be undefined for us. I'll throw in a check and that'll throw an error if it is.